DOCSP-39732 Run a command standardization (#77)
mongodb/docs-java-rs: source/write/run-command.txt

.. _java-rs-run-command:

======================
Run a Database Command
======================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: database operation, code example

Overview
--------

> In this guide, you can learn how to run a database command with the
> {+driver-short+}. Database commands can perform a variety of administrative and

Reviewer: ```suggestion
In this guide, you can learn how to run a database command by using the
{+driver-short+}. You can use database commands to perform a variety of administrative and
```

diagnostic tasks, such as fetching server statistics, initializing a replica
set, or running an aggregation pipeline. You can use database commands when the
driver does not contain a helper method for the command you want to run.

> .. important:: Prefer Driver Methods Over Database Commands

Reviewer: S: so that you dont stack two important admonitions, maybe this one can be a tip instead

   The driver provides methods for many database commands. We recommend

>    using driver methods instead of executing database commands when possible. 

Reviewer: S: remove this last sentence as the first part of the note expresses a similar note

   To perform administrative tasks, use the `MongoDB Shell

>    <https://www.mongodb.com/docs/mongodb-shell/>`__ instead of the

Reviewer: S: use the shell role https://github.com/mongodb/snooty-parser/blob/5dab215d105fa01976c6521c356323725e137a9d/snooty/rstspec-consolidated.toml#L1359C7-L1361

   {+driver-short+}. Calling the MongoDB Shell :manual:`db.runCommand()
   </reference/method/db.runCommand/>` method is the preferred method to issue
   database commands, as it provides a consistent interface between the shell
   and drivers.

.. include:: includes/reactor-note.rst

Run a Command
-------------

To run a database command, specify the command in a document, and pass the
document to the ``runCommand()`` method. The following code calls the
``runCommand()`` method on a database to run the ``ping`` command, which is a
no-op command used to test if a server is responsive.

.. io-code-block::

   .. input:: /includes/write/run-command.java

>       :language:  Java

Reviewer: ```suggestion
      :language: java
```

      :emphasize-lines: 1, 3

   .. output::
      :visible: false

      Document{{ok=1}}

Additional Information
----------------------

> To see a full list of database commands and their available parameters, see the :manual:`Database
> Commands </reference/command>` in the {+mdb-server+} manual.
> 

Reviewer: ```suggestion
To see a full list of database commands and their available parameters, see :manual:`Database
Commands </reference/command>` in the {+mdb-server+} manual.

```

API Documentation
~~~~~~~~~~~~~~~~~

To learn more about any of the methods or types discussed in this
guide, see the following API documentation:

> - `runCommand() <{+api+}/mongodb-driver-reactivestreams/com/mongodb/reactivestreams/client/MongoDatabase.html#runCommand(org.bson.conversions.Bson)>`__

Reviewer: S: since theres only one list item, maybe you can convert this into a sentence.
